util/iosutil/iosutil_test.go

```diff
@@ -0,0 +1,149 @@
+package iosutil
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestParseVersion(t *testing.T) {
+	tests := []struct {
+		description     string
+		given           string
+		expectedVersion Version
+		expectedError   string
+	}{
+		{
+			description:     "Valid",
+			given:           "14.2",
+			expectedVersion: Version{Major: 14, Minor: 2},
+		},
+		{
+			description:   "Invalid Parts - Empty",
+			given:         "",
+			expectedError: "expected major.minor format",
+		},
+		{
+			description:   "Invalid Parts - Too Few",
+			given:         "14",
+			expectedError: "expected major.minor format",
+		},
+		{
+			description:   "Invalid Parts - Too Many",
+			given:         "14.2.1",
+			expectedError: "expected major.minor format",
+		},
+		{
+			description:   "Invalid Major",
+			given:         "xxx.2",
+			expectedError: "major version is not an integer",
+		},
+		{
+			description:   "Invalid Minor",
+			given:         "14.xxx",
+			expectedError: "minor version is not an integer",
+		},
+	}
+
+	for _, test := range tests {
+		version, err := ParseVersion(test.given)
+
+		if test.expectedError == "" {
+			assert.NoError(t, err, test.description+":err")
+		} else {
+			assert.EqualError(t, err, test.expectedError, test.description+":err")
+		}
+
+		assert.Equal(t, test.expectedVersion, version, test.description+":version")
+	}
+}
+
+func TestEqualOrGreater(t *testing.T) {
+	givenMajor := 14
+	givenMinor := 2
+
+	tests := []struct {
+		description  string
+		givenVersion Version
+		expected     bool
+	}{
+		{
+			description:  "Less Than By Major + Minor",
+			givenVersion: Version{Major: 13, Minor: 1},
+			expected:     false,
+		},
+		{
+			description:  "Less Than By Major",
+			givenVersion: Version{Major: 13, Minor: 2},
+			expected:     false,
+		},
+		{
+			description:  "Less Than By Minor",
+			givenVersion: Version{Major: 14, Minor: 1},
+			expected:     false,
+		},
+		{
+			description:  "Equal",
+			givenVersion: Version{Major: 14, Minor: 2},
+			expected:     true,
+		},
+		{
+			description:  "Greater By Major + Minor",
+			givenVersion: Version{Major: 15, Minor: 3},
+			expected:     true,
+		},
+		{
+			description:  "Greater By Major",
+			givenVersion: Version{Major: 15, Minor: 2},
+			expected:     true,
+		},
+		{
+			description:  "Greater By Minor",
+			givenVersion: Version{Major: 14, Minor: 3},
+			expected:     true,
+		},
+	}
+
+	for _, test := range tests {
+		result := test.givenVersion.EqualOrGreater(givenMajor, givenMinor)
+		assert.Equal(t, test.expected, result, test.description)
+	}
+}
+
+func TestDetectVersionGroup(t *testing.T) {
```

Comment:
Nitpick: Can we change this test name to `TestDetectVersionClassification` to more closely match the function under test?

Reply:
Ah, yes. Good eye. I renamed the method during a refactor and forgot to update the corresponding test.